src/hash.cr

```diff
@@ -171,6 +171,36 @@ class Hash(K, V)
     indexes.map {|index| self[index] }
   end
 
+  # Returns the first key with the given *value*, else raises `KeyError`.
+  #
+  # ```
+  # hash = {"foo": "bar", "baz": "qux"}
+  # hash.key("bar")    #=> "foo"
+  # hash.key("qux")    #=> "baz"
+  # hash.key("foobar") #=> Missing hash key for value: foobar (KeyError)
+  # ```
+  def key(value)
+    each do |k, v|
+      return k if v == value
+    end
+    raise KeyError.new "Missing hash key for value: #{value}"
+  end
+
+  # Returns *true* if any key pairs with the given *value*.
+  #
+  # ```
+  # hash = {"foo": "bar", "baz": "qux"}
+  # hash.key("bar")    #=> true
+  # hash.key("qux")    #=> true
+  # hash.key("foobar") #=> false
+  # ```
+  def key?(value)
+    each do |k, v|
+      return true if v == value
+    end
```

Comment:
Looking at the contract here, I wonder if this should be implemented similarly to `fetch`, have one overload that takes a block and the other overloads be implemented in terms of it. Even if we decide against making the block variant public API, this should be DRY'ed up with a private `key_internal` at very least.